public/pages/DetectorResults/containers/AnomalyResultsTable.tsx

```diff
@@ -78,6 +78,14 @@ export function AnomalyResultsTable(props: AnomalyResultsTableProps) {
       ? props.anomalies.filter((anomaly) => anomaly.anomalyGrade > 0)
       : [];
 
+        // Helper function to properly quote values for rison encoding
+  const quoteRisonValue = (value: string): string => {
+    if (value.includes(' ') || value.includes(',') || value.includes('(') || value.includes(')') || value.includes('!') || value.includes(':') || value.includes("'")) {
```

Comment:
Should we also include characters such as `*`, `\t`, `\r`, `"` or `%` that can break the URL‑encoded rison string?  Can we rely on `rison.encode()` (or `encode_rison_component`) instead of manual quoting?

Reply:
Those are not considered as special characters. Only the ones I included in the helper function will cause encoding issue

Reply:
I tried to use 

{"value": 10.24, "timestamp": "2020-01-01 00:01:00", "host": "O'Brien"}

here is what I got

<img width="1900" height="875" alt="Screenshot 2025-07-24 at 5 11 40 PM" src="https://github.com/user-attachments/assets/0303cbb0-084a-4875-8eff-877979eef562" />

Reply:
I was reading a wrong issue about ribon encoding, it only listed out the characters that I initially included. Updated the implementation to use the `setStateToOsdUrl` provided by OSD to build the url - https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/plugins/opensearch_dashboards_utils/public/state_management/url/osd_url_storage.ts#L95